src/backend/marsha/core/models/video.py

```diff
@@ -424,6 +430,25 @@ class LiveRegistration(BaseModel):
         verbose_name=_("LTI consumer site"),
     )
 
+    email = models.EmailField(_("email address"), db_index=True)
+
+    initial_type = models.CharField(
+        max_length=20,
+        verbose_name=_("created by which type of action"),
+        help_text=_("created by which type of action"),
+        choices=REGISTRATION_INITIAL_TYPE_CHOICES,
+        default=REGISTRATION,
+        null=True,
+        blank=True,
+    )
```

Comment:
Why is this necessary?
The attendance is materialized by the fact that `live_attendance` records some time stamps
The registration is materialized by the presence of an object in database with no materialized attendance or could be materialized by just a flag `is_registered`?

Reply:
the flag is_registered would be another idea that has the same result IMO, but with my proposition we can extend the log system to other types

Reply:
I would like to see what others think but to me the name of the flag "initial_type" does not convey the information that "the user was registered". I understand that it was initially like this but that it may have changed afterward... it is confusing.

Reply:
I had trouble finding the right term, I was looking for a word meaning how it got created originally. I requested Nico to brainstorm on the name with me, we came out with these different ones initiated_by, created_by_type, created_by (not mad because it makes me expect a username)
Do you prefer any of this term, but I can go for the the field is_registered if you prefer

Reply:
I may miss something here, but, I don't get in which case `initial_type` is set to `ATTENDANCE`.
I think answering this could help us to find a better name.

Reply:
ATTENDANCE will be set when a record is created for a user watching a live that wasn't registered. A field is_registered set to TRUE or FALSE is enough in that matter. We don't know if we would need to set this field later on to other value, such as STARTED (if video has already started for instance). With the logs contained in live_attendance we will be able to know everything we can keep it simple and just tag if the user is registered or not, it seems clearer and enough.
I will then put is_registered to false as a default value and we will have to update the existing record to true.

Reply:
So, if I understand well, initial_type is useless for now ?

Reply:
yes it's to prepare the PR after, In the introduction of a feature that will monitor attendances

Reply:
So, forcing `REGISTRATION` in the serializer validator is temporary I presume ?
If it is, maybe add a comment about it ?

Reply:
I've updated the model with a field is_registered to make things simpler.